include/TS3.h

```diff
@@ -3,27 +3,36 @@
 
 #include <iostream>
 
-#include "TDetector.h"
+#include "TGRSIDetector.h"
 #include "TS3Hit.h"
 
-class TS3 : public TDetector {
+class TS3 : public TGRSIDetector {
 	public:
 		TS3();
-		~TS3();
+		TS3(const TS3&);
+		virtual  ~TS3();
 
-		void AddFragment(TFragment*, MNEMONIC*);
-		void BuildHits() {} //no need to build any hits, everything already done in AddFragment
+		virtual void AddFragment(TFragment*, MNEMONIC*);
+		virtual void BuildHits();
 
+		TGRSIDetectorHit* GetHit(const int& idx =0);
 		TS3Hit* GetS3Hit(const int& i);  
-		Short_t GetMultiplicity() { return fS3Hits.size(); }
+		Short_t GetMultiplicity() const { return fS3Hits.size(); }
+		void PushBackHit(TGRSIDetectorHit* deshit);
 
-		TVector3 GetPosition(int front, int back);
+		static TVector3 GetPosition(int ring, int sector);
 
-      virtual void Clear(Option_t *opt = "all");		     //!
-      virtual void Print(Option_t *opt = "") const;		  //!
+		void Copy(TObject&) const;
+		TS3& operator=(const TS3&);  // 
+		virtual void Clear(Option_t *opt = "all");		     //!
+		virtual void Print(Option_t *opt = "") const;		  //!
+		
+		
 
 	private:
 		std::vector<TS3Hit> fS3Hits;
+		std::vector<TFragment*> fS3_RingFragment; //! 
+		std::vector<TFragment*> fS3_SectorFragment; //! 
```

Comment:
Multiplicity two event form coulex would be well past the unsafe mark
unless you have a very thin and light target.

 since you have been down this road before, I am sure that you know you can
fairly easily decide between collected charge and timing which events go
together.

On Wed, Nov 4, 2015 at 9:18 PM, jsmallcombe <notifications@github.com>
wrote:

> In include/TS3.h
> <https://github.com/GRIFFINCollaboration/GRSISort/pull/584#discussion_r43969498>
> :
>
> >
> >  	private:
> >  		std::vector<TS3Hit> fS3Hits;
> > +		std::vector<TFragment*> fS3_RingFragment; //!
> > +		std::vector<TFragment*> fS3_SectorFragment; //!
>
> Time difference is fine if the events are sequential, but in a genuine 2
> hit event the timing will be almost indistinguishable. But that is an
> analysis issue rather than a fundamental or data processing which I believe
> we should be focusing on here.
>
> I'm just trying to write good S3 code, this isnt my first DSSD rodeo. We
> are mostly focused on coulex at the moment, but the events you want arent
> the only ones in your detector. And depending on masses you can have good
> multiplicity 2 events in coulex.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GRIFFINCollaboration/GRSISort/pull/584/files#r43969498>
> .
>